src/api/app/controllers/webui/request_controller.rb

```diff
@@ -284,10 +284,14 @@ def delete_request
 
     rescue APIException => e
       flash[:error] = e.message
-      redirect_to :controller => :package, :action => :show, :package => params[:package], :project => params[:project] and return if params[:package]
-      redirect_to :controller => :project, :action => :show, :project => params[:project] and return
+      if params[:package]
+        redirect_to package_show_path(project: params[:project], package: params[:package])
+      else
+        redirect_to project_show_path(project: params[:project])
+      end
+      return
     end
-    redirect_to :controller => :request, :action => :show, :number => req.number
+    redirect_to request_show_path(number: req.number)
```

Comment:
I would move this line inside the `begin... rescue` block and then you can get rid of the `return`